### PR TITLE
#1544 Separate cluster creation of zookeeper

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1544)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/ClusterInfo.scala [hunk 1]
```diff
@@ -57,5 +57,9 @@ trait ClusterInfo {
     * @param newNodeNames new node names
     * @return an new instance
     */
+  //TODO remove this after we finished #1544
   def clone(newNodeNames: Set[String]): ClusterInfo
+
+  def clone2(state: Option[String], error: Option[String]): ClusterInfo =
```
Comment (chia7712) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/ClusterInfo.scala]: please add more document for this workaround

Comment (saivirtue) [reply ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/ClusterInfo.scala]: OK

**Comment by chia7712** [comment]
@saivirtue Is there a issue to remind UI developer to update the APIs?

**Comment by saivirtue** [comment]
> @saivirtue Is there a issue to remind UI developer to update the APIs?

I think so. Since zookeeper NEED to be created in two APIs.
May recall required front-end developers to join this PR.

**Comment by chia7712** [comment]
> I think so. Since zookeeper NEED to be created in two APIs.
May recall required front-end developers to join this PR after pass backend QA.

@konekoya @wu87988622 we bring a party to you :)

**Comment by chia7712** [comment]
@saivirtue 理想上zk, bk, wk and streamapp的creation and update應該會有很多類似的判斷檢查，可否麻煩你抽象一層json refiner給所有creation/update作為基礎? 這樣可以避免不同cluster對類似參數產生不同的檢查邏輯

上述可以在獨立的PR完成 或是你方便就在此pr先做一部分

**Comment by saivirtue** [comment]
> @saivirtue 理想上zk, bk, wk and streamapp的creation and update應該會有很多類似的判斷檢查，可否麻煩你抽象一層json refiner給所有creation/update作為基礎? 這樣可以避免不同cluster對類似參數產生不同的檢查邏輯
> 
> 上述可以在獨立的PR完成 或是你方便就在此pr先做一部分

OK, let me abstract part of jsonRefiner class for zookeeper in this PR.

**Comment by saivirtue** [comment]
retry

**Comment by saivirtue** [comment]
/cc @wu87988622 
thanks for reporting the issue of zookeeper api in k8s env.
please testing this pr after the QA pass :)

**Pull request merged**